internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -1411,6 +1411,22 @@ func (h *Handler) createCs3Share(ctx context.Context, w http.ResponseWriter, r *
 		}
 	}
 
+	expiry := r.PostFormValue("expireDate")
+	if expiry != "" {
+		ts, err := time.Parse("2006-01-02T15:04:05-0700", expiry)
+		if err != nil {
+			return nil, &ocsError{
+				Code:    response.MetaBadRequest.StatusCode,
+				Message: "could not parse expiry timestamp on this item",
+				Error:   err,
+			}
+		}
+		req.Grant.Expiration = &types.Timestamp{
+			Seconds: uint64(ts.Unix()),
+			Nanos:   0,
+		}
```

Comment:
There is `utils.TimeToTS` which could do this for you. But not critical.

Reply:
Thx, did not know that, incorporated it.